### PR TITLE
feat: enable resume download for hf_hub_download

### PR DESCRIPTION
**Pull request by lonelam**
enable auto resume download by default to prevent always restart on downloading large file with poor network connection.

### CODE REVIEW

File: private_gpt/settings/settings.py [hunk 1]
```diff
@@ -92,6 +92,7 @@ class LocalSettings(BaseModel):
     llm_hf_repo_id: str
     llm_hf_model_file: str
     embedding_hf_model_name: str
+    llm_hf_resume_download: bool
```
Comment (lopagela) [inline private_gpt/settings/settings.py:95]: Suggesting to make it default to `False` if no values are provided in documentation
```suggestion
    llm_hf_resume_download: bool = False
```

File: settings.yaml [hunk 2]
```diff
@@ -30,6 +30,7 @@ local:
   llm_hf_repo_id: TheBloke/Mistral-7B-Instruct-v0.1-GGUF
   llm_hf_model_file: mistral-7b-instruct-v0.1.Q4_K_M.gguf
   embedding_hf_model_name: BAAI/bge-small-en-v1.5
+  llm_hf_resume_download: True
```
Comment (lopagela) [inline settings.yaml:33]: Preferring json style boolean (even though `True` is [supported](https://yaml.org/type/bool.html)) as it stay consistent with the `false` in this file.

```suggestion
  llm_hf_resume_download: true
```

**Review by lopagela — COMMENTED** [review]
Suggesting to make it `False` by default in `settings.py` to do not explicitly require it in the chain of `settings*.yaml`

**Review by pabloogc — CHANGES_REQUESTED** [review]
I think going for a very simple solution here is way better. Just making it always resume the download seems like a healthy default.

That way you don't have to overcomplicate with a setting that realistically nobody will need to change.

**Comment by lonelam** [comment]
@pabloogc , thank you for your review and suggestions. I agree that resuming downloads by default is a prudent approach. However, to address scenarios where resuming may not be ideal, such as when the model file in the HuggingFace repository has been updated, I propose adding an optional setting in the LocalSettings class. This would allow users to choose whether to resume downloads from the last breakpoint or start afresh, thereby ensuring flexibility and robustness in handling different situations.

**Comment by pabloogc** [comment]
> @pabloogc , thank you for your review and suggestions. I agree that resuming downloads by default is a prudent approach. However, to address scenarios where resuming may not be ideal, such as when the model file in the HuggingFace repository has been updated, I propose adding an optional setting in the LocalSettings class. This would allow users to choose whether to resume downloads from the last breakpoint or start afresh, thereby ensuring flexibility and robustness in handling different situations.

I still wouldn't expose it as a setting, instead pass a parameter to the setup script `scripts/setup --resume=true/false` defaulting to true should be enough for 99.9% of scenarios.

File: private_gpt/settings/settings.py (hunk 1 above)
Comment (lonelam) [reply private_gpt/settings/settings.py]: Has been removed from settings.

File: settings.yaml (hunk 2 above)
Comment (lonelam) [reply settings.yaml]: The option has been removed from settings.

**Review by pabloogc — APPROVED** [review]

**Pull request merged**